[2024-08-15T00:38:07.133+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: data_pipeline_api.analyze_data scheduled__2024-08-14T00:30:00+00:00 [queued]>
[2024-08-15T00:38:07.139+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: data_pipeline_api.analyze_data scheduled__2024-08-14T00:30:00+00:00 [queued]>
[2024-08-15T00:38:07.143+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 2
[2024-08-15T00:38:07.154+0000] {taskinstance.py:1380} INFO - Executing <Task(BashOperator): analyze_data> on 2024-08-14 00:30:00+00:00
[2024-08-15T00:38:07.159+0000] {standard_task_runner.py:57} INFO - Started process 43193 to run task
[2024-08-15T00:38:07.161+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'data_pipeline_api', 'analyze_data', 'scheduled__2024-08-14T00:30:00+00:00', '--job-id', '48', '--raw', '--subdir', 'DAGS_FOLDER/spark-jobs.py', '--cfg-path', '/tmp/tmpgep1_xa6']
[2024-08-15T00:38:07.164+0000] {standard_task_runner.py:85} INFO - Job 48: Subtask analyze_data
[2024-08-15T00:38:07.235+0000] {task_command.py:415} INFO - Running <TaskInstance: data_pipeline_api.analyze_data scheduled__2024-08-14T00:30:00+00:00 [running]> on host 27ef58cc88fc
[2024-08-15T00:38:07.327+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='data_pipeline_api' AIRFLOW_CTX_TASK_ID='analyze_data' AIRFLOW_CTX_EXECUTION_DATE='2024-08-14T00:30:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-08-14T00:30:00+00:00'
[2024-08-15T00:38:07.329+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2024-08-15T00:38:07.331+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'bash /opt/***/jobs/spark-submit.sh /opt/***/jobs/analyze_data.py ']
[2024-08-15T00:38:07.341+0000] {subprocess.py:86} INFO - Output:
[2024-08-15T00:38:07.354+0000] {subprocess.py:93} INFO - /opt/***/jobs/analyze_data.py
[2024-08-15T00:38:07.454+0000] {subprocess.py:93} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-08-15T00:38:08.948+0000] {subprocess.py:93} INFO - 24/08/15 00:38:08 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-08-15T00:38:09.652+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SparkContext: Running Spark version 3.5.1
[2024-08-15T00:38:09.653+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SparkContext: OS info Linux, 6.6.32-linuxkit, aarch64
[2024-08-15T00:38:09.653+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SparkContext: Java version 11.0.23
[2024-08-15T00:38:09.671+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO ResourceUtils: ==============================================================
[2024-08-15T00:38:09.672+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-08-15T00:38:09.672+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO ResourceUtils: ==============================================================
[2024-08-15T00:38:09.672+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SparkContext: Submitted application: FinanceDataAnalysis
[2024-08-15T00:38:09.694+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 1024, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-08-15T00:38:09.704+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO ResourceProfile: Limiting resource is cpus at 1 tasks per executor
[2024-08-15T00:38:09.707+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-08-15T00:38:09.748+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SecurityManager: Changing view acls to: ***
[2024-08-15T00:38:09.748+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SecurityManager: Changing modify acls to: ***
[2024-08-15T00:38:09.748+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SecurityManager: Changing view acls groups to:
[2024-08-15T00:38:09.749+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SecurityManager: Changing modify acls groups to:
[2024-08-15T00:38:09.749+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-08-15T00:38:09.906+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO Utils: Successfully started service 'sparkDriver' on port 33047.
[2024-08-15T00:38:09.939+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SparkEnv: Registering MapOutputTracker
[2024-08-15T00:38:09.962+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SparkEnv: Registering BlockManagerMaster
[2024-08-15T00:38:09.974+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-08-15T00:38:09.975+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-08-15T00:38:09.977+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-08-15T00:38:09.989+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-12da018b-f75d-4ba3-89eb-12734c6f44ad
[2024-08-15T00:38:09.996+0000] {subprocess.py:93} INFO - 24/08/15 00:38:09 INFO MemoryStore: MemoryStore started with capacity 1458.4 MiB
[2024-08-15T00:38:10.005+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-08-15T00:38:10.074+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-08-15T00:38:10.103+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2024-08-15T00:38:10.108+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO Utils: Successfully started service 'SparkUI' on port 4041.
[2024-08-15T00:38:10.127+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO SparkContext: Added JAR file:///opt/bitnami/spark/resources/elasticsearch-spark-30_2.12-8.4.3.jar at spark://27ef58cc88fc:33047/jars/elasticsearch-spark-30_2.12-8.4.3.jar with timestamp 1723682289646
[2024-08-15T00:38:10.177+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2024-08-15T00:38:10.212+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO TransportClientFactory: Successfully created connection to spark-master/172.18.0.4:7077 after 14 ms (0 ms spent in bootstraps)
[2024-08-15T00:38:10.300+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20240815003810-0035
[2024-08-15T00:38:10.309+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 40211.
[2024-08-15T00:38:10.310+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO NettyBlockTransferService: Server created on 27ef58cc88fc:40211
[2024-08-15T00:38:10.311+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-08-15T00:38:10.317+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 27ef58cc88fc, 40211, None)
[2024-08-15T00:38:10.319+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO BlockManagerMasterEndpoint: Registering block manager 27ef58cc88fc:40211 with 1458.4 MiB RAM, BlockManagerId(driver, 27ef58cc88fc, 40211, None)
[2024-08-15T00:38:10.321+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 27ef58cc88fc, 40211, None)
[2024-08-15T00:38:10.322+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 27ef58cc88fc, 40211, None)
[2024-08-15T00:38:10.445+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO StandaloneAppClient$ClientEndpoint: Master removed worker worker-20240809124721-172.18.0.7-36933: Not receiving heartbeat for 60 seconds
[2024-08-15T00:38:10.453+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO StandaloneSchedulerBackend: Worker worker-20240809124721-172.18.0.7-36933 removed: Not receiving heartbeat for 60 seconds
[2024-08-15T00:38:10.453+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO StandaloneAppClient$ClientEndpoint: Master removed worker worker-20240809124721-172.18.0.6-35359: Not receiving heartbeat for 60 seconds
[2024-08-15T00:38:10.454+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO StandaloneSchedulerBackend: Worker worker-20240809124721-172.18.0.6-35359 removed: Not receiving heartbeat for 60 seconds
[2024-08-15T00:38:10.458+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO TaskSchedulerImpl: Handle removed worker worker-20240809124721-172.18.0.7-36933: Not receiving heartbeat for 60 seconds
[2024-08-15T00:38:10.459+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO TaskSchedulerImpl: Handle removed worker worker-20240809124721-172.18.0.6-35359: Not receiving heartbeat for 60 seconds
[2024-08-15T00:38:10.461+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO DAGScheduler: Shuffle files lost for worker worker-20240809124721-172.18.0.7-36933 on host 172.18.0.7
[2024-08-15T00:38:10.463+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO DAGScheduler: Shuffle files lost for worker worker-20240809124721-172.18.0.6-35359 on host 172.18.0.6
[2024-08-15T00:38:10.485+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-08-15T00:38:10.641+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-08-15T00:38:10.642+0000] {subprocess.py:93} INFO - 24/08/15 00:38:10 INFO SharedState: Warehouse path is 'file:/tmp/***tmp1bye35rt/spark-warehouse'.
[2024-08-15T00:38:11.339+0000] {subprocess.py:93} INFO - 24/08/15 00:38:11 INFO Version: Elasticsearch Hadoop v8.4.3 [a7873eb469]
[2024-08-15T00:38:13.200+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2024-08-15T00:38:13.201+0000] {subprocess.py:93} INFO -   File "/opt/***/jobs/analyze_data.py", line 48, in <module>
[2024-08-15T00:38:13.201+0000] {subprocess.py:93} INFO -     save_analysis_result(top_gainers, "data/top_gainers.csv")
[2024-08-15T00:38:13.202+0000] {subprocess.py:93} INFO -   File "/opt/***/jobs/analyze_data.py", line 41, in save_analysis_result
[2024-08-15T00:38:13.203+0000] {subprocess.py:93} INFO -     df.coalesce(1).write.mode("overwrite").csv(file_path, header=True)
[2024-08-15T00:38:13.204+0000] {subprocess.py:93} INFO -   File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 1864, in csv
[2024-08-15T00:38:13.204+0000] {subprocess.py:93} INFO -   File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2024-08-15T00:38:13.205+0000] {subprocess.py:93} INFO -   File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2024-08-15T00:38:13.212+0000] {subprocess.py:93} INFO - pyspark.errors.exceptions.captured.AnalysisException: [UNSUPPORTED_DATA_TYPE_FOR_DATASOURCE] The CSV datasource doesn't support the column `_metadata` of the type "MAP<STRING, STRING>".
[2024-08-15T00:38:13.261+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO SparkContext: Invoking stop() from shutdown hook
[2024-08-15T00:38:13.261+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-08-15T00:38:13.266+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO SparkUI: Stopped Spark web UI at http://27ef58cc88fc:4041
[2024-08-15T00:38:13.269+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-08-15T00:38:13.271+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-08-15T00:38:13.280+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-08-15T00:38:13.287+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO MemoryStore: MemoryStore cleared
[2024-08-15T00:38:13.287+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO BlockManager: BlockManager stopped
[2024-08-15T00:38:13.290+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-08-15T00:38:13.290+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-08-15T00:38:13.296+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO SparkContext: Successfully stopped SparkContext
[2024-08-15T00:38:13.296+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO ShutdownHookManager: Shutdown hook called
[2024-08-15T00:38:13.296+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-ee9b35b4-ec93-4196-ad79-985cc53458f9
[2024-08-15T00:38:13.300+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-ee9b35b4-ec93-4196-ad79-985cc53458f9/pyspark-2862e285-aa07-4df5-9ddf-7369caa196af
[2024-08-15T00:38:13.304+0000] {subprocess.py:93} INFO - 24/08/15 00:38:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-52c92ef7-3d96-45d3-937d-553a3faff174
[2024-08-15T00:38:13.329+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2024-08-15T00:38:13.341+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/bash.py", line 210, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2024-08-15T00:38:13.344+0000] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=data_pipeline_api, task_id=analyze_data, execution_date=20240814T003000, start_date=20240815T003807, end_date=20240815T003813
[2024-08-15T00:38:13.353+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 48 for task analyze_data (Bash command failed. The command returned a non-zero exit code 1.; 43193)
[2024-08-15T00:38:13.357+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-08-15T00:38:13.372+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
